libsolidity/interface/FileReader.h

```diff
@@ -61,18 +61,17 @@ class FileReader
 	void allowDirectory(boost::filesystem::path _path);
 	FileSystemPathSet const& allowedDirectories() const noexcept { return m_allowedDirectories; }
 
-	StringMap const& sourceCodes() const noexcept { return m_sourceCodes; }
-
-	/// Retrieves the source code for a given source unit name.
-	SourceCode const& sourceCode(SourceUnitName const& _sourceUnitName) const { return m_sourceCodes.at(_sourceUnitName); }
+	/// @returns all sources by their internal source unit names.
+	StringMap const& sourceUnits() const noexcept { return m_sourceCodes; }
 
 	/// Resets all sources to the given map of source unit name to source codes.
 	/// Does not enforce @a allowedDirectories().
-	void setSources(StringMap _sources);
+	void setSourceUnits(StringMap _sources);
```

Comment:
I think we should remove this method and add this instead:
```c++
void updateSourceUnit(std::string const& _sourceUnitName, SourceCode _source);
```

Reply:
You have to decide, would you prefer `update` based on the file name or on the source unit name?

Reply:
I think update should work in both cases. Adding a new entry should only work based on file name.

Reply:
I won't do the refactor with the constructor as it currently really does not work.